edk2-rockchip/Platform/Rockchip/Rk356x/AcpiTables/Mshc.asl

```diff
@@ -33,4 +33,54 @@ Device (MSH0) {
         })
         Return (RBUF)
     }
-}
\ No newline at end of file
+}
+
+// Mobile Storage Host Controller
+Device (MSH1) {
+    Name (_HID, "PRP0001")
+    Name (_UID, 4)
+    Name (_CCA, Zero)
+
+    Name (_DSD, Package () {
+        ToUUID("daffd814-6eba-4d8c-8a91-bc9bbf4aa301"),
+        Package () {
+            Package () { "compatible", Package () { "rockchip,rk3568-dw-mshc", "rockchip,rk3288-dw-mshc" } },
+            Package () { "fifo-depth", 0x100 },
+            Package () { "max-frequency", 50000000 },
+            Package () { "bus-width", 4 },
+            Package () { "cap-sd-highspeed", 1 },
+#if FixedPcdGetBool(PcdMshc1SdioIrq) == 1
+            Package () { "cap-sdio-irq", 1 },
+#endif
+            Package () { "disable-wp", 1 },
+#if FixedPcdGetBool(PcdMshc1NonRemovable) == 1
+            Package () { "non-removable", 1 }
+#endif
+        }
+    })
+
+    Method (_CRS, 0x0, Serialized) {
+        Name (RBUF, ResourceTemplate() {
+            Memory32Fixed (ReadWrite, 0xFE2C0000, 0x4000)
+            Interrupt (ResourceConsumer, Level, ActiveHigh, Exclusive) { 131 }
+        })
+        Return (RBUF)
+    }
+
+#if FixedPcdGetBool(PcdMshc1Enable) == 0
+    Method (_STA) {
+        Return (0x0) // disabled
```

Comment:
Return 0xF if it is enabled.

Reply:
Yes, but this is the other way around. If _STA is not defined, it's assumed that the devices exists. To disable a device, _STA needs to be fined and return 0x0. In this case I include MSHC1 in the Quartz64/SOQuartz, but keep it disabled by having _STA return 0. Or should I just put #if FixedPcdGetBool(PcdMshc1Enable) == 1 around the whole MSHC1 block?

Reply:
I guess putting the Enable check around the whole block would make it more clear that the device then "doesn't exist", hence the _STA can be removed. That will probably make the whole block clearer. What do you think?

Reply:
BBR mandates _STA for all device nodes. I have to go back and add it for all of the other ones too.

Reply:
Ok, then I'll provide _STA which returns 0xF and enclose the whole device with that if. Just pushed an updated patchiest.

Reply:
Ok, as discussed I'll provide Name (_STA, FixedPcdGet8(PcdMshc1Status)). That seems pretty nice. Pushed a patch set with it.